jobs/anac-certified-attributes-importer/src/service/processor.ts

```diff
@@ -0,0 +1,218 @@
+  const jobCorrelationId = crypto.randomUUID()
+  const batchSize = recordsBatchSize
+
+  const fileContent = await sftpClient.downloadCSV()
+
+  const attributes: AnacAttributes = await getAttributesIdentifiers(readModel, anacTenantId)
+
+  const preparedProcessTenants = processTenants(tenantProcess, refreshableToken, attributes, jobCorrelationId)
+
+  let scanComplete = false
+  let fromLine = 1
+
+  do {
+    const batchResult: BatchParseResult = getBatch(fileContent, fromLine, batchSize, jobCorrelationId)
+
+    // TODO Not sure this is the best way to satisfy type checks
+    const paOrgs: PaRow[] = batchResult.records
+      .map((org: CsvRow) => {
+        if ('codice_ipa' in org && org.codice_ipa) return org as PaRow
+        else return null
+      })
+      .filter((r): r is PaRow => r !== null)
+
+    const nonPaOrgs: NonPaRow[] = batchResult.records
+      .map((org: CsvRow) => {
+        if ('codice_ipa' in org) return null
+        else return org
+      })
+      .filter((r): r is NonPaRow => r !== null)
+    //
+
+    await preparedProcessTenants(
+      paOrgs,
+      (org) => org.codice_ipa,
+      (codes) => readModel.getPATenants(codes)
+    )
+    await preparedProcessTenants(
+      nonPaOrgs,
+      (org) => org.cf_gestore,
+      (codes) => readModel.getNonPATenants(codes)
+    )
+
+    fromLine = fromLine + batchSize
+    scanComplete = batchResult.processedRecordsCount === 0
+  } while (!scanComplete)
+}
+
+async function getAttributesIdentifiers(readModel: ReadModelQueries, anacTenantId: string): Promise<AnacAttributes> {
+  const anacTenant: PersistentTenant = await readModel.getTenantById(anacTenantId)
+  const certifier = anacTenant.features.find((f) => f.type === 'PersistentCertifier')
+
+  if (!certifier) {
+    throw Error(`Tenant with id ${anacTenantId} is not a certifier`)
+  }
+
+  const anacAbilitato = await readModel.getAttributeByExternalId(certifier.certifierId, ANAC_ABILITATO_CODE)
+  const anacIncaricato = await readModel.getAttributeByExternalId(certifier.certifierId, ANAC_INCARICATO_CODE)
+  const anacInConvalida = await readModel.getAttributeByExternalId(certifier.certifierId, ANAC_IN_CONVALIDA_CODE)
+
+  return {
+    anacAbilitato: { id: anacAbilitato.id, externalId: { origin: anacAbilitato.origin, value: anacAbilitato.code } },
+    anacIncaricato: {
+      id: anacIncaricato.id,
+      externalId: { origin: anacIncaricato.origin, value: anacIncaricato.code },
+    },
+    anacInConvalida: {
+      id: anacInConvalida.id,
+      externalId: { origin: anacInConvalida.origin, value: anacInConvalida.code },
+    },
+  }
+}
+
+const processTenants =
+  (
+    tenantProcess: TenantProcessService,
+    refreshableToken: RefreshableInteropToken,
+    attributes: AnacAttributes,
+    jobCorrelationId: string
+  ) =>
+  async <T extends CsvRow>(
+    orgs: T[],
+    extractTenantCode: (org: T) => string,
+    retrieveTenants: (codes: string[]) => Promise<PersistentTenant[]>
+  ): Promise<void> => {
```

Comment:
I try to avoid declaring totally anonymous functions. This is because if an error happens in there, the stacktrace will not be able to identify the the function which the error was thrown.

Ex: 
```javascript
const testA = () => () => {
  throw new Error("Hello");
};

const a = testA();
a();
```
Stacktrace:
```bash
Error: Hello
    at /Users/carmineporricelli/Desktop/deleteMe.js:2:9
    at Object.<anonymous> (/Users/carmineporricelli/Desktop/deleteMe.js:6:1)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
```

If you instead return the function giving it a name:
```
const testA = () => {
  return function testB() {
    throw new Error("Hello");
  };
};
```
The stacktrace will be able to tell you more precisely where the error was thrown.
```
Error: Hello
    at testB (/Users/carmineporricelli/Desktop/deleteMe.js:3:11)
    at Object.<anonymous> (/Users/carmineporricelli/Desktop/deleteMe.js:8:1)
```

Reply:
nice, I didn't know! 
Fixed, thanks ❤️